pkg/apiroutes/version.go

```diff
@@ -0,0 +1,144 @@
+	EnvResponse struct {
+		Version        string `json:"codewind_version"`
+		ImageBuildTime string `json:"image_build_time"`
+	}
+)
+
+// GetContainerVersions : Get the versions of each Codewind container, for a given connection ID
+func GetContainerVersions(conID string, httpClient utils.HTTPClient) (ContainerVersions, error) {
+	conInfo, conInfoErr := connections.GetConnectionByID(conID)
+	if conInfoErr != nil {
+		return ContainerVersions{}, conInfoErr.Err
+	}
+
+	var containerVersions ContainerVersions
+	PFEVersion, err := GetPFEVersionFromConnection(conInfo, http.DefaultClient)
+	if err != nil {
+		return ContainerVersions{}, err
+	}
+
+	GatekeeperVersion, err := GetGatekeeperVersionFromConnection(conInfo, http.DefaultClient)
+	if err != nil {
+		return ContainerVersions{}, err
+	}
+
+	PerformanceVersion, err := GetPerformanceVersionFromConnection(conInfo, http.DefaultClient)
+	if err != nil {
+		return ContainerVersions{}, err
+	}
+
+	// Get cwctl version from inside the code
+	containerVersions.CwctlVersion = appconstants.VersionNum
+	containerVersions.PFEVersion = PFEVersion
+	containerVersions.GatekeeperVersion = GatekeeperVersion
+	containerVersions.PerformanceVersion = PerformanceVersion
+
+	return containerVersions, nil
+}
+
+// GetPFEVersionFromConnection : Get the version of the PFE container, deployed to the connection with the given ID
+func GetPFEVersionFromConnection(connection *connections.Connection, HTTPClient utils.HTTPClient) (string, error) {
+	req, err := http.NewRequest("GET", connection.URL+"/api/v1/environment", nil)
+	if err != nil {
+		return "", err
+	}
+
+	version, err := getVersionFromEnvAPI(req, connection, HTTPClient)
+	if err != nil {
+		return "", err
+	}
+	return version, err
+}
+
+// GetGatekeeperVersionFromConnection : Get the version of the Gatekeeper container, deployed to the connection with the given ID
+func GetGatekeeperVersionFromConnection(connection *connections.Connection, HTTPClient utils.HTTPClient) (string, error) {
+	req, err := http.NewRequest("GET", connection.URL+"/api/v1/gatekeeper/environment", nil)
+	if err != nil {
+		return "", err
+	}
+
+	version, err := getVersionFromEnvAPI(req, connection, HTTPClient)
+	if err != nil {
+		return "", err
+	}
+	return version, err
+}
+
+// GetPerformanceVersionFromConnection : Get the version of the Performance container, deployed to the connection with the given ID
+func GetPerformanceVersionFromConnection(connection *connections.Connection, HTTPClient utils.HTTPClient) (string, error) {
+	req, err := http.NewRequest("GET", connection.URL+"/performance/api/v1/environment", nil)
+	if err != nil {
+		return "", err
+	}
+
+	version, err := getVersionFromEnvAPI(req, connection, HTTPClient)
+	if err != nil {
+		return "", err
+	}
+	return version, err
+}
+
+func getVersionFromEnvAPI(req *http.Request, connection *connections.Connection, HTTPClient utils.HTTPClient) (string, error) {
+	client := &http.Client{}
```

Comment:
http client is being passed in as a function parameter,  seems we dont need to create a NEW client here.   removing this line would also allow us to send in a mock client when writing tests.

Reply:
Amazing, hopefully this is why I was having problems mocking for the tests